rand_test.go

```diff
@@ -12,3 +15,19 @@ func TestRand(t *testing.T) {
 		t.Fatal(err)
 	}
 }
+
+func TestAllocations(t *testing.T) {
+	n := int(testing.AllocsPerRun(10, func() {
+		buf := make([]byte, 32)
+		openssl.RandReader.Read(buf)
+		sink ^= buf[0]
+	}))
+	want := 1
+	ver := strings.TrimPrefix(runtime.Version(), "devel ")
+	if version.Compare(ver, "go1.24") >= 0 {
+		want = 0
+	}
```

Comment:
I anticipate we'll want to bring this out to a utility file if we add more of these in the future. Not sure it's worth doing it proactively, though, if we don't expect more.

Either way, might be worth having a comment pointing at cgo_go124.go to explain what this is all about.

Reply:
Will do both, thanks!